tests/Fetcher/TululaFetcherTest.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Tests\Fetcher;
+
+use App\Entity\Conference;
+use App\Entity\Continent;
+use App\Entity\Tag;
+use App\Fetcher\LocationGuesser;
+use App\Fetcher\TululaFetcher;
+use App\Repository\ExcludedTagRepository;
+use App\Repository\TagRepository;
+use Prophecy\Argument;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+
+class TululaFetcherTest extends KernelTestCase
+{
+    private function createFetcher(MockResponse $response, ?string $selectedTag = 'php'): TululaFetcher
+    {
+        $locationGuesser = $this->prophesize(LocationGuesser::class);
+        $locationGuesser
+            ->getContinent(Argument::type('string'))
+            ->willReturn($continent = new Continent());
+        $continent->setName('Europe');
+        $continent->setEnabled(true);
+
+        $tagRepository = $this->prophesize(TagRepository::class);
+        $tagRepository
+            ->findTagByName(Argument::type('string'))
+            ->willReturn($tag = new Tag());
+        $tag->setName($selectedTag);
+        $tag->setSelected(true);
+
+        $excludedTagRepository = $this->prophesize(ExcludedTagRepository::class);
+        $excludedTagRepository
+            ->findOneBy(Argument::type('array'))
+            ->willReturn(null);
+        $client = new MockHttpClient($response);
+
+        $fetcher = new TululaFetcher(
+            $locationGuesser->reveal(),
+            $tagRepository->reveal(),
+            $excludedTagRepository->reveal(),
+            $client
+        );
+
+        return $fetcher;
+    }
+
+    /**
+     * @dataProvider provideConferences
+     */
+    public function testFetch(array $fakeConference, ?string $selectedTag = 'php')
+    {
+        $data = json_decode(file_get_contents('tests/Fetcher/fixtures.json'), true);
+        $data['data']['events']['events'] = $fakeConference;
+
+        $response = new MockResponse(json_encode($data));
```

Comment:
Why do json_decode + json_encode?
This is really strange what you did here. You mixed 2 concept : manual fixture and raw fixtures. Use first or second but not both at the same time in the very same test.

Here, you could only use manual fixture.

**BUT** I would like another test with a real raw HTTP response